runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -517,34 +536,26 @@ public void delete(@ConflictParameter K key) {
 
     /** {@inheritDoc} */
     @Override
-    @Mutator(name = "putAll",
-            undoFunction = "undoPutAll",
-            undoRecordFunction = "undoPutAllRecord",
-            conflictParameterFunction = "putAllConflictFunction")
+    @TransactionalMethod
     public void putAll(@Nonnull Map<? extends K, ? extends V> m) {
-        // If we have no index functions, then just directly put all
-        if (secondaryIndexes.isEmpty()) {
-            mainMap.putAll(m);
-        } else {
-            // Otherwise we must update all secondary indexes
-            // TODO: Do this in parallel (need to acquire update locks, potentially)
-            m.entrySet().stream()
-                    .forEach(e -> {
-                        V previous = mainMap.put(e.getKey(), e.getValue());
-                        unmapSecondaryIndexes(e.getKey(), previous);
-                        mapSecondaryIndexes(e.getKey(), e.getValue());
-                    });
-        }
+        m.entrySet().stream().forEach(entry -> put(entry.getKey(), entry.getValue()));
     }
 
     /** {@inheritDoc} */
     @Override
-    @Mutator(name = "clear", reset = true)
+    @Mutator(name = "clear", garbageIdentifyFunction = "identifyClearGarbage", reset = true)
     public void clear() {
         mainMap.clear();
         secondaryIndexes.values().forEach(Map::clear);
     }
 
+    @DontInstrument
+    protected List<Object> cleanGarbage(Object locator) {
```

Comment:
Why this is not used in CorfuTable$CORFUSMR?

Reply:
Ah..I have modified cleanGarbage to identifyClearGarbage. 
done.